exporter/collector/testdata/fixtures/gke_control_plane_metrics_agent_metrics_expect.json

```diff
@@ -1062,5 +1062,49 @@
         }
       ]
     }
+  ],
+  "selfObservabilityMetrics": [
```

Comment:
> Any new OC metrics would cause this to fail

Yes and running the regeneration script would add any new metrics. Is there a different approach you'd suggest?

> It only sees OC metrics instantiated as part of our integration test setup

Yes which includes our instrumentation in observability.go and exporterhelper in the collector repo. There could be more.

> We'll need to rework the mechanism to pull self-observability metrics in the future, but we plan to always map it to this abstraction.

Which self-observability metrics do you mean here?

Reply:
> Yes and running the regeneration script would add any new metrics. Is there a different approach you'd suggest?

(Ugh.. i deleted my previous comment...😑)
Is there a public/internal doc that has those steps documented? Things like how was the script triggered? Is it possible to automate the script so that it can sync up with the latest status

Reply:
Ya, it's documented here https://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/main/exporter/collector/integrationtests.md#adding-new-testshttps://github.com/GoogleCloudPlatform/opentelemetry-operations-go/blob/main/exporter/collector/integrationtests.md#adding-new-tests. The idea is to manually run the script when you make an intentional change or add a new test. We shouldn't need to automate since the tests should fail before it goes out of sync.